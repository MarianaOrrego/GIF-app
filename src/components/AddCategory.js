import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ( e ) => {
    setInputValue(e.target.value);

    console.log('handleInputChange');
  }

  const handleSubmit = ( e ) => {
    e.preventDefault();

    if ( inputValue.trim().length > 2) {
      setCategories( cate => [inputValue, ...cate] );
      setInputValue('');
    }

  }

  return (
    <form onSubmit={ handleSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

