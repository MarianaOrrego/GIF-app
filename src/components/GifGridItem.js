import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

  return (
    <div className="cardgif animate__animated animate__fadeIn animate__delay-2s">
        <img src={ url } alt={ title } />
        <p> { title } </p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

