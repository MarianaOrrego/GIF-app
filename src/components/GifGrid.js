import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGird = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn animate__delay-1s">{category}</h3>

            { loading && <p className="animate__animated animate__flash animate__delay-1s">Loading</p> }

            <div className="cardgrid">

                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGird.propTypes = {
    category: PropTypes.string.isRequired
}