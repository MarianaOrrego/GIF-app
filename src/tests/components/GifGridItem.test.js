import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'Titulo';
    const url = 'https//hola.jpg'
    const wrapper = shallow( <GifGridItem title = { title } url = { url }/>);

    test('Debe mostrar el componenete correctamente ', () => {
      
        expect (wrapper).toMatchSnapshot();
    })
    
    test('Debe tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( title );
    })
    
    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );
    })
    
    test('Debe tener animate__fadeIn', () => {
      
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true)
    })
    
})
