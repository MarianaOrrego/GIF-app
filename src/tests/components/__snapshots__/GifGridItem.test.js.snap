// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componenete correctamente  1`] = `
<div
  className="cardgif animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="Titulo"
    src="https//hola.jpg"
  />
  <p>
     
    Titulo
     
  </p>
</div>
`;
